client/src/redux/sagas/intro.sagas.js

```diff
@@ -19,10 +45,21 @@ export function* watchFetchIntroMessages() {
   yield takeLeading('FETCH_INTRO', fetchIntroMessages);
 }
 
+// Write IntroMessages to the database and update the redux store.
 export function* saveIntroMessages(action) {
   yield put({ type: 'INTRO_SAVE_STARTED' });
   try {
-    localStorage.setItem('introMessages', JSON.stringify(action.payload));
+    const introMessages = {};
+    Object.entries(action.payload).forEach(([component, unread]) => {
+      introMessages[component] = { unread };
+    });
+
+    yield call(fetchUtil, {
+      url: '/api/intromessages/',
+      method: 'PUT',
+      body: JSON.stringify(introMessages)
```

Comment:
When saving an intromessage, what is the reasoning behind sending the entire state, instead of just the updated message state?

Reply:
With futures changes planned to IntroMessages, I minimized the amount of work on the front end.  Because of this I chose not to change the way the redux store worked so I left it working the way IntroMessages worked with local storage.